src/Bridge/Doctrine/Persister/ObjectManagerPersister.php

```diff
@@ -62,7 +62,7 @@ public function persist($object)
             // Check if the ID is explicitly set by the user. To avoid the ID to be overridden by the ID generator
             // registered, we disable it for that specific object.
             if ($metadata instanceof ORMClassMetadataInfo) {
-                if ($metadata->usesIdGenerator() && false === empty($metadata->getIdentifierValues($object))) {
+                if ($metadata->usesIdGenerator() && false === empty($metadata->getIdentifierValues($object)) && !$metadata->idGenerator instanceof IdGenerator) {
```

Comment:
Could you also change: `0 !== count($metadata->getIdentifierValues($object))`? I prefer to avoid using `empty()`

Reply:
Done